sklearn/neighbors/_base.py

```diff
@@ -553,12 +567,24 @@ def _fit(self, X, y=None):
                 **self.effective_metric_params_,
             )
         elif self._fit_method == "kd_tree":
-            self._tree = KDTree(
-                X,
-                self.leaf_size,
-                metric=self.effective_metric_,
-                **self.effective_metric_params_,
-            )
+            if (
+                self.effective_metric_ == "minkowski"
+                and self.effective_metric_params_.get("w") is not None
+            ):
+                raise ValueError(
+                    "algorithm='kd_tree' is not valid for "
+                    "metric='minkowski' with a weight parameter 'w': "
+                    "try algorithm='ball_tree' "
+                    "or algorithm='brute' instead."
+                )
+
+            else:
```

Comment:
Nit: I prefer to remove the `else` here, so there is less indentation. (especially when the first `if` statements are for error checking)